src/lcfiplus.cc

```diff
@@ -89,7 +89,11 @@ const Vertex* Event::getPrimaryVertex(const char* privtxname) const {
   if(primaryVertices.size() > 0 ){
     return primaryVertices[0];
   } else {
-    return nullptr;
+	// Primary vertex is used for adapting vertex z-position deviation from 0 and thus always requied.
+	// Our best estimation would be position(0.,0.,0.) and errors{0.,0.,0.,0.,0.,beam spot size in z}.
+	double dz = static_cast<double>(Globals::Instance()->getBeamSizeZ());
+    std::unique_ptr<Vertex> dummy = std::make_unique<Vertex>(0.,0.,0.,0.,dz,static_cast<const double*>(0),false);
+    return dummy.get();
```

Comment:
I think this construct leads to a dangling pointer once the function returns, because the `unique_ptr<Vertex>` goes out of scope then and cleans up the resources it owns, leaving the pointer returned by `dummy.get()` pointing to a deleted `Vertex` now.

Unfortunately `std::unique_ptr` is templated also on its deleter, so since the other part of the function returns a managed pointer, it seems like it is not easily possible to use that here. What you could do is use a `std::shared_ptr` instead, since that does not have it's deleter in its type. Something along the lines of
```cpp
std::shared_ptr<const Vertex> Event::getPrimaryVertex(const char* privtxname) const {
  // Everything that needs to be done until here
  // ...
  if (primaryVertices.size() > 0) {
    // Give this shared_ptr a deleter that does nothing when it is called
    return std::shared_ptr<const Vertex*>(primaryVertices[0], [](const Vertex*){ /* vertex is managed for us */});
  } else {
    // Here we use the default deleter, so that this Vertex is managed for us
    return std::make_shared<const Vertex>(0.,0.,0.,0.,dz,static_cast<const double*>(0),false);
  }
}
```

This is not extremely elegant, but it would get around the problem of having to manage the resources slightly differently depending on whether there is a primary vertex or not. Another solution could be to always return a copy of the primary vertex and make the caller manage it (in that case one could return a `unique_ptr`).

Maybe the easiest solution would be to check for the presence of a primary vertex even earlier, when the internal `primaryVertices` collection is filled from the LCIO contents, and simply add the dummy vertex there(?). That would make the situation in this function a bit simpler to deal with, I think.

Reply:
In case the caller does not delete the returned vertex on could probably also do sth. like:
```
static Vertex defaultVertex(0.,0.,0.,0.,dz,static_cast<const double*>(0),false);
///...
return &defaultVertex ;
```